app/client/components/RegionFocusSwitcher.ts

```diff
@@ -0,0 +1,667 @@
+    const gristDoc = this._getGristDoc();
+    if (gristDoc && region?.type === 'panel' && region?.id === 'main') {
+      throw new Error('main panel is not supported when a view layout is rendered');
+    }
+    if (!gristDoc && region?.type === 'section') {
+      throw new Error('view section id is not supported when no view layout is rendered');
+    }
+
+    this._state.set({region, initiator: options.initiator});
+  }
+
+  public focusActiveSection() {
+    const gristDoc = this._getGristDoc();
+    if (gristDoc) {
+      this.focusRegion({type: 'section', id: gristDoc.viewModel.activeSectionId()});
+    }
+  }
+
+  public reset(initiator?: StateUpdateInitiator) {
+    this.focusRegion(undefined, {initiator});
+  }
+
+  public panelAttrs(id: Panel, ariaLabel: string) {
+    return [
+      dom.attr('role', 'region'),
+      dom.attr('aria-label', ariaLabel),
+      dom.attr(ATTRS.regionId, id),
+      dom.cls('kb-focus-highlighter-group', use => {
+        use(this._initiated);
+        // highlight focused elements everywhere except in the grist doc views
+        if (id !== 'main') {
+          return true;
+        }
+        const gristDoc = this._gristDocObs ? use(this._gristDocObs) : null;
+        if (!gristDoc) {
+          return true;
+        }
+        if (gristDoc) {
+          use(gristDoc.activeViewId);
+        }
+        return isSpecialPage(gristDoc);
+      }),
+      dom.cls('clipboard_group_focus', use => {
+        use(this._initiated);
+        const gristDoc = this._gristDocObs ? use(this._gristDocObs) : null;
+        // if we are not on a grist doc, whole page is always focusable
+        if (!gristDoc) {
+          return true;
+        }
+        const current = use(this._state).region;
+        // on a grist doc, panel content is focusable only if it's the current region
+        if (current?.type === 'panel' && current.id === id) {
+          return true;
+        }
+        if (gristDoc) {
+          use(gristDoc.activeViewId);
+        }
+        // on a grist doc, main panel is focusable only if we are not the actual document view
+        return isSpecialPage(gristDoc);
+      }),
+      dom.cls('clipboard_forbid_focus', use => {
+        use(this._initiated);
+        // forbid focus only on the main panel when on an actual document view (and not a special page)
+        if (id !== 'main') {
+          return false;
+        }
+        const gristDoc = this._gristDocObs ? use(this._gristDocObs) : null;
+        if (!gristDoc) {
+          return false;
+        }
+        if (gristDoc) {
+          use(gristDoc.activeViewId);
+        }
+        if (isSpecialPage(gristDoc)) {
+          return false;
+        }
+        return true;
+      }),
+      dom.cls(`${cssFocusedPanel.className}-focused`, use => {
+        use(this._initiated);
+        const current = use(this._state);
+        return current.initiator?.type === 'cycle' && current.region?.type === 'panel' && current.region.id === id;
+      }),
```

Comment:
in this, I use the `cls` helper multiple times to apply different classes. But the content of each callback is rather similar, they mostly subscribe to the same observables.

Is there a way for me to write only one callback to apply different classes? I feel this could end up running way less code each time an observable changes.

Reply:
Oh, to respond to this, yes, you can apply different classes by giving a callback as the first argument, and having the callback return the class name, e.g. `dom.cls(use => isSpecialPage(use(gristDoc.activeViewId)) ? 'class1' : 'class2')`.